src/test/java/org/takes/http/BkBasicTest.java

```diff
@@ -370,10 +367,38 @@ public void run() {
      * @return Prepared Socket mock
      * @throws IOException If some problem inside
      */
-    private static Socket createMockSocket() throws IOException {
-        final Socket socket = Mockito.mock(Socket.class);
-        Mockito.when(socket.getInputStream()).thenReturn(
-            new ByteArrayInputStream(
+    private static BkBasicTest.MkSocket createMockSocket() throws IOException {
+        return new BkBasicTest.MkSocket();
+    }
+
+    /**
+     * Socket mock for reuse.
+     */
+    private static final class MkSocket extends Socket {
```

Comment:
@essobedo please make it a standalone class in test sources, it can be reused later on